source/Halibut/Transport/SecureListener.cs

```diff
@@ -287,7 +287,7 @@ async Task ExecuteRequest(TcpClient client)
                 finally
                 {
                     SafelyRemoveClientFromTcpClientManager(client, clientName);
-                    SafelyCloseStream(stream, clientName);
+                    await SafelyCloseStreamAsync(stream, clientName);
                     SafelyCloseClient(client, clientName);
```

Comment:
No async close method exists on the TCPClient.

The test in `sast/write-a-test-to-find-sync-io` suggest that with these change sync IO is not done on the stream. So this is probably fine.